src/zenml/steps/base_step.py

```diff
@@ -163,14 +164,17 @@ def __new__(
         # tfx requires them to be unique
         # TODO [ENG-155]: Can we prefix inputs and outputs to avoid this
         #  restriction?
-        shared_input_output_keys = set(cls.INPUT_SIGNATURE).intersection(
-            set(cls.OUTPUT_SIGNATURE)
-        )
-        if shared_input_output_keys:
+        all_keys = list(cls.INPUT_SIGNATURE) + list(cls.OUTPUT_SIGNATURE)
```

Comment:
I guess it's not super relevant in this case because the list of all keys will never be too large, but I'd still suggest using a `collections.Counter` object to count the occurrences of keys to make it a little more expressive.

Reply:
@schustmi thank you for the tip. I have added a new version utilizing the `Counter`.